File: backend/education/src/main/java/com/braintrust/education/application/dtos/dtos/CourseDTO.java
package com.braintrust.education.application.dtos.dtos;


import java.time.LocalDateTime;
import java.util.List;

public record CourseDTO(
        String id,
        String code,
        String name,
        String description,
        String urlImage,
        String grade,
        String group,
        String teacherId,
        String teacherName,
        boolean active,
        int studentCount,
        int assignmentCount,
        int unitCount,
        LocalDateTime createdAt
) {}

File: frontend/app/shared/dtos/course.dto.ts
export interface CourseDTO {
    id: string;
    code: string;
    name: string;
    description: string;
    urlImage: string;
    grade: string;
    group: string;
    teacherId: string;
    teacherName: string;
    active: boolean;
    studentCount: number;
    assignmentCount: number;
    unitCount: number;
    createdAt: string;
}

export interface CourseUnitDTO {
    id: string;
    courseId: string;
    name: string;
    urlImage: string;
    numUnity: number;
    description: string;
}

export interface CourseStatsDTO {
    totalCourses: number;
    activeCourses: number;
    inactiveCourses: number;
    totalStudents: number;
    totalTeachers: number;
    averageStudentsPerCourse: number;
}

export interface PaginatedResponseDTO<T> {
    content: T[];
    pageNumber: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    first: boolean;
    last: boolean;
}